lib/src/copies.rs

```diff
@@ -121,28 +121,26 @@ impl Stream for CopiesTreeDiffStream<'_> {
     type Item = CopiesTreeDiffEntry;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        self.inner.as_mut().poll_next(cx).map(|option| {
-            option.map(|diff_entry| {
-                let Some(CopyRecord { source, .. }) =
-                    self.copy_records.for_target(&diff_entry.path)
-                else {
-                    return CopiesTreeDiffEntry {
-                        source: diff_entry.path.clone(),
-                        target: diff_entry.path,
-                        value: diff_entry.value,
-                    };
-                };
-
-                CopiesTreeDiffEntry {
-                    source: source.clone(),
-                    target: diff_entry.path,
-                    value: diff_entry.value.and_then(|(_, target_value)| {
-                        self.source_tree
-                            .path_value(source)
-                            .map(|source_value| (source_value, target_value))
-                    }),
-                }
-            })
-        })
+        let Some(diff_entry) = ready!(self.inner.as_mut().poll_next(cx)) else {
```

Comment:
This looked quite confusing to me until I read that `ready!` can return early. That's a bit surprising to me (it feels like it should be called `return_unless_ready!` or something. I suppose I'll get used to it, but it's very confusing until you know what it does.

So I wonder if it's worth it, which I guess depends on what this commit does. I didn't understand that either. I figured the two `.map()` calls (on `task::Poll` and on `Option`) would have the same effect. What's the difference?

Reply:
The reason behind this change is to wrap the whole block within a `while let _ = ready!(..)` loop. Alternatively, we can use `loop { match self.inner.as_mut().poll_next(cx) { .. return; .. }; .. }`.

Perhaps, `ready!()` would make sense before when `try!(..)` was a thing.

Reply:
> The reason behind this change is to wrap the whole block within a `while let _ = ready!(..)` loop. Alternatively, we can use `loop { match self.inner.as_mut().poll_next(cx) { .. return; .. }; .. }`.

I see. The "return early" made me think we currently did some unnecessary processing.

> Perhaps, `ready!()` would make sense before when `try!(..)` was a thing.

Yes, that's what I thought after I learnt about the hidden early return.

Reply:
Updated commit message and switched to `std::task::ready!()`.